python/cudf_polars/cudf_polars/experimental/parallel.py

```diff
@@ -127,12 +155,32 @@ def task_graph(
         return graph, (key_name, 0)
 
 
+def get_client():
+    """Get appropriate Dask client or scheduler."""
+    SerializerManager.register_serialize()
+
+    try:  # pragma: no cover; block depends on executor type and Distributed cluster
+        from distributed import get_client
+
+        client = get_client()
+        SerializerManager.run_on_cluster(client)
+    except (
+        ImportError,
+        ValueError,
+    ):  # pragma: no cover; block depends on Dask local scheduler
+        from dask import get
+
+        return get
+    else:  # pragma: no cover; block depends on executor type and Distributed cluster
+        return client.get
+
+
 def evaluate_dask(ir: IR) -> DataFrame:
     """Evaluate an IR graph with Dask."""
-    from dask import get
-
     ir, partition_info = lower_ir_graph(ir)
 
+    get = get_client()
+
```

Comment:
TODO: in a followup, allow configuration of which executor is used (independently of whether an implicit client exists).

Reply:
Just to clarify are you suggesting we support a case where in the same process the user may execute tasks with the Dask local scheduler _and_ a `distributed.Client`?

Reply:
Yes, I know dask likes this implicit selection, but I don't

Reply:
>Yes, I know dask likes this implicit selection, but I don't

I also don't like the way `client` detection is expected to be an automatic/magic process in Dask. I don't think it's intuitive at all, and I doubt the "convenience" outweighs the confusion it sometimes causes.

With that said, I'm fairly confident that we will end up causing even more confusion if we break from Dask norms before we are certain that we have a better end-to-end design in mind.